src/main/scala/edu/berkeley/ce/rockslicing/BlockVTK.scala

```diff
@@ -0,0 +1,213 @@
+  }
+
+  /**
+    * Finds the indices of the face vertices in the global list of vertices
+    * @param faceOrientedVerts Mapping from each face to a Seq of vertices for that face
+    * @param vertices Seq of distinct vertices for the rock block
+    * @return A mapping from each face of the block to a Seq of integers that represent
+    *         the indices of the vertices in the global vertex list
+    */
+  private def connectivity(faceOrientedVerts: Map[Face, Seq[(Double, Double, Double)]],
+                           vertices: Seq[(Double, Double, Double)]): Map[Face, Seq[Int]] = {
+    faceOrientedVerts.keys.zip(
+      faceOrientedVerts map { case (_, orientedVertices) =>
+        orientedVertices map { vertex =>
+          vertices.indexOf(vertex)
+        }
+      }
+    ).toMap
+  }
+
+  /**
+    * Creates a sequence of all the normals of all the faces for the list of input blocks
+    * @param faces A mapping from each face of the block to a Seq of vertices for that face arranged in counter-
+    *        clockwise order relative to its unit normal
+    * @return Sequence of tuples representing normals of all the block faces
+    */
+  private def normals(faces: Seq[Face]): Seq[(Double, Double, Double)] = {
+    faces map { case face => (face.a, face.b, face.c) }
+  }
+
+  /**
+    * Determines the offset that defines each face in the connectivity list
+    * @param connectivities Mapping from each face of a block to a Seq of integers that represent
+    *        the indices of the vertices in the global vertex list
+    * @return A Seq of integers that represent the offset of each face in the
+    *         connectivity list
+    */
+  private def faceOffsets(connectivities: Map[Face, Seq[Int]]): Seq[Int] = {
+    val localOffsets = connectivities map { case(_, connections) => connections.length }
+    // val offsets = Seq[Int](localOffsets.head)
+    val offsets = Seq[Int]()
+
+    def offsetIterator(globalOS: Seq[Int], localOS: Seq[Int]): Seq[Int] = {
+      localOS match {
+        case Nil => globalOS.reverse
+        case offset +: rest => if (globalOS.isEmpty) {
+          offsetIterator(Seq[Int](offset), rest)
+        } else {
+          offsetIterator((offset + globalOS.head) +: globalOS, rest)
+        }
+      }
+    }
+    offsetIterator(offsets, localOffsets.toSeq)
+  }
+}
+
+/**
+  * Simple data structure to contain data that represents a rock block that can be turned into vtk format
+  * @param Rock block 
+  * @constructor Create a new rock block in format that can be turned into vtk by rockProcessor
+  */
+case class BlockVTK(block: Block) {
+  val orientedVertices = BlockVTK.orientVertices(block)
+  val tupleVertices = orientedVertices.values.flatten.toSeq.distinct
+  val connectivityMap = BlockVTK.connectivity(orientedVertices, tupleVertices)
+  val vertices = tupleVertices flatMap { t =>
+    List(t._1, t._2, t._3)
+  }
+  val vertexIDs = tupleVertices.indices
+  val connectivity = connectivityMap.values.flatten
```

Comment:
It's taking the `Seq[Int]` portion of the map that represents the connectivity for each face and flattening it into a single `Seq[Int]`. It needs to be in this format for `pyevtk`. `offsets` is then used to specify which chunks of `connectivity` correspond to what faces. That's just how it's dealt with in `vtk` format. Does that make sense?